### PR TITLE
Split CI into per-workspace jobs with path filtering (#64)

### LINKED ISSUE #64: Split CI into per-workspace jobs with path filtering
## Context

CI currently runs four flat jobs (lint, test, test-ct, build) that delegate to all workspaces via root scripts. With three workspaces now carrying their own test suites (485 stagebook + 59 viewer + 136 vscode = 680 tests), splitting into per-workspace test jobs gives faster feedback, clearer failure attribution, and the ability to skip unaffected jobs via path filtering.

Also: CI triggers still list the deleted `vscode-extension` branch — needs cleanup.

### 1. Remove stale branch trigger

Drop `vscode-extension` from the `on.push.branches` and `on.pull_request.branches` lists in `.github/workflows/ci.yml`.

### 2. Split test job into per-workspace jobs

Replace the single `Unit Tests (vitest)` job with three:

| Job | Workspace | Command | Depends on |
|-----|-----------|---------|------------|
| `test-stagebook` | `packages/stagebook` | `npm test -w stagebook` | — |
| `test-viewer` | `apps/viewer` | `npm run build -w stagebook && npm test -w stagebook-viewer` | — |
| `test-vscode` | `apps/vscode` | `npm run build -w stagebook && npm test -w stagebook-vscode` | — |

Note: viewer and vscode import stagebook via workspace link, so they need `npm run build -w stagebook` first to produce the `dist/` artifacts their tests resolve against.

### 3. Add path filtering

Use `paths` filters on each job so changes to one workspace don't trigger unrelated test jobs:

| Job | Triggers on changes to |
|-----|----------------------|
| `test-stagebook` | `packages/stagebook/**` |
| `test-viewer` | `apps/viewer/**`, `packages/stagebook/**` |
| `test-vscode` | `apps/vscode/**`, `packages/stagebook/**` |
| `test-ct` (Playwright) | `packages/stagebook/**` |
| `lint` | `**/*.ts`, `**/*.tsx`, `**/*.js`, `**/*.json` (keep broad — lint is cheap) |
| `build` | all paths (keep broad — catches packaging regressions) |

The key rule: **a change to `packages/stagebook/**` triggers all test jobs**, because viewer and vscode consume it. A change to only `apps/viewer/**` skips stagebook and vscode tests.

### 4. Keep existing jobs unchanged

- **Lint & Format** — stays root-level (`npm run lint && npm run format:check`). Fast, catches cross-workspace issues.
- **Component Tests (Playwright)** — stays stagebook-only (`npm run test:ct -w stagebook`).
- **Build** — stays root-level (`npm run build`). Catches packaging / export regressions across all workspaces.

## File to modify

`.github/workflows/ci.yml` — single file change.

## Consumer usage

No consumer-facing change — this is internal CI infrastructure. PR check names change from a single "Unit Tests (vitest)" to three per-workspace checks, giving clearer signal on which workspace broke.

## Edge cases to verify

- A PR touching only `apps/viewer/src/` runs `test-viewer` but NOT `test-stagebook` or `test-vscode`
- A PR touching only `packages/stagebook/src/` runs ALL three test jobs + Playwright CT
- A PR touching only `.github/workflows/ci.yml` runs all jobs (workflow change = full run)
- A PR touching only `docs/` or `README.md` skips all test jobs (no path match)

## Acceptance criteria

- `vscode-extension` removed from branch triggers
- Three separate test jobs appear in PR checks
- Path filtering works as described above (verify by opening a viewer-only PR and confirming stagebook/vscode tests are skipped)
- All jobs pass on the PR that implements this change
- Total CI wall-clock time for a full run is not worse than today (parallel jobs should improve it)

### PR DESCRIPTION
**Pull request by JamesPHoughton**
## Summary

- Drop the deleted `vscode-extension` branch from the `on.push`/`on.pull_request` triggers in `.github/workflows/ci.yml`.
- Replace the single "Unit Tests (vitest)" job with three per-workspace jobs: `test-stagebook`, `test-viewer`, `test-vscode`.
- Add a `changes` detection job (using `dorny/paths-filter@v3`) and gate each test job's `if:` on its output, so unaffected workspaces are skipped.
- Viewer and vscode jobs still `npm run build -w stagebook` first, because they import stagebook via the workspace link and resolve against `dist/`.

Closes #64.

## Path filter matrix

| Job | Triggers on changes to |
|-----|----------------------|
| `test-stagebook` | `packages/stagebook/**`, `.github/workflows/ci.yml` |
| `test-viewer` | `apps/viewer/**`, `packages/stagebook/**`, `.github/workflows/ci.yml` |
| `test-vscode` | `apps/vscode/**`, `packages/stagebook/**`, `.github/workflows/ci.yml` |
| `test-ct` (Playwright) | `packages/stagebook/**`, `.github/workflows/ci.yml` |
| `lint` | `**/*.ts`, `**/*.tsx`, `**/*.js`, `**/*.json`, `.github/workflows/ci.yml` |
| `build` | always (no filter — catches packaging regressions) |

`.github/workflows/ci.yml` is included in every filter so workflow edits force a full run.

## Test plan

- [x] `npm run format:check` passes locally.
- [ ] On this PR (which touches only `.github/workflows/ci.yml`), all jobs run — verifies the "workflow change = full run" behavior.
- [ ] A follow-up viewer-only PR would run `test-viewer` + `build` + `lint` only — skipping `test-stagebook`, `test-vscode`, and `test-ct`. (Not exercised here; verifiable once merged.)
- [ ] A follow-up docs-only PR would skip all test jobs + lint; only `build` runs. (Not exercised here.)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,13 +2,51 @@ name: CI
...
+    name: Detect changes
+    runs-on: ubuntu-latest
+    outputs:
+      stagebook: ${{ steps.filter.outputs.stagebook }}
+      viewer: ${{ steps.filter.outputs.viewer }}
+      vscode: ${{ steps.filter.outputs.vscode }}
+      ct: ${{ steps.filter.outputs.ct }}
+      lint: ${{ steps.filter.outputs.lint }}
+    steps:
+      - uses: actions/checkout@v4
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply .github/workflows/ci.yml:20]: Applied — added `fetch-depth: 0` to the `changes` job's checkout in d68123a. Thanks for the catch.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,13 +2,51 @@ name: CI
...
+              - 'packages/stagebook/**'
+              - '.github/workflows/ci.yml'
+            ct:
+              - 'packages/stagebook/**'
+              - '.github/workflows/ci.yml'
+            lint:
+              - '**/*.ts'
+              - '**/*.tsx'
+              - '**/*.js'
+              - '**/*.json'
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply .github/workflows/ci.yml:54]: Partially applied in d68123a: added root `package.json`, `package-lock.json` to every workspace filter (so dependency-only PRs re-run all workspace tests), and `.prettierrc` to the `lint` filter.

I deliberately didn't include per-workspace `tsconfig*.json`, `vitest.config.*`, `playwright*.config.*`, or `vite.config.*` because those already live under `packages/stagebook/**`, `apps/viewer/**`, or `apps/vscode/**` and are already matched by the workspace-scoped filters. Keeping the filter lists focused on only the truly root-level files that aren't otherwise covered.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by JamesPHoughton** [comment]
Ready for review — @jamesphoughton

---
_Generated by [Claude Code](https://claude.ai/code)_
